src/main/java/graphql/execution/instrumentation/fieldvalidation/FieldValidationEnvironment.java

```diff
@@ -0,0 +1,50 @@
+package graphql.execution.instrumentation.fieldvalidation;
+
+import graphql.GraphQLError;
+import graphql.PublicApi;
+import graphql.execution.ExecutionContext;
+import graphql.execution.ExecutionPath;
+
+import java.util.Map;
+
+/**
+ * This contains all of the field and their arguments for a given query.  The method
+ * {@link #getFields()} will be where most of the useful validation information is
+ * contained.  It also gives you a helper to make validation error messages.
+ *
+ * @see FieldAndArguments
+ */
+@PublicApi
+public interface FieldValidationEnvironment {
```

Comment:
One advantage is for consumers in THEIR testing since they can more easily mock an interface than a value class.

Also in this case I had helper methods (mkError) and that is nicer behind an interface I think

I think we should err towards interfaces in general

Reply:
my thoughts:

Interfaces are the more cleaner way to represent an API. The mocking argument is also true (if you are not using Spock or want to mock it by hand).

The only downside with interfaces is that it is a bit more overhead, but I think in the long term interfaces are more safer for an library to expose.

So ... sure: I agree to favor interfaces.